src/backend/distributed/utils/statistics_collection.c

```diff
@@ -128,28 +114,31 @@ ClusterSize(List *distributedTableList)
 {
 	uint64 clusterSize = 0;
 	ListCell *distTableCacheEntryCell = NULL;
+	MemoryContext savedContext = CurrentMemoryContext;
 
-	foreach(distTableCacheEntryCell, distributedTableList)
+	PG_TRY();
 	{
-		DistTableCacheEntry *distTableCacheEntry = lfirst(distTableCacheEntryCell);
-		Oid relationId = distTableCacheEntry->relationId;
-		MemoryContext savedContext = CurrentMemoryContext;
-
-		PG_TRY();
+		foreach(distTableCacheEntryCell, distributedTableList)
 		{
-			Datum distTableSizeDatum = DirectFunctionCall1(citus_table_size,
-														   ObjectIdGetDatum(relationId));
+			DistTableCacheEntry *distTableCacheEntry = lfirst(distTableCacheEntryCell);
+			Oid relationId = distTableCacheEntry->relationId;
+			Datum distTableSizeDatum =
+				DirectFunctionCall1(citus_table_size, ObjectIdGetDatum(relationId));
 			clusterSize += DatumGetInt64(distTableSizeDatum);
 		}
-		PG_CATCH();
-		{
-			FlushErrorState();
-
-			/* citus_table_size() throws an error while the memory context is changed */
-			MemoryContextSwitchTo(savedContext);
-		}
-		PG_END_TRY();
 	}
+	PG_CATCH();
+	{
+		ErrorData *edata = NULL;
+		MemoryContextSwitchTo(savedContext);
+		edata = CopyErrorData();
+		FlushErrorState();
+
+		/* rethrow as WARNING */
+		edata->elevel = WARNING;
+		ThrowErrorData(edata);
+	}
+	PG_END_TRY();
```

Comment:
And also, can we add a regression test that triggers this function? Currently, the enterprise-master Travis builds fails due to low coverage. It might help to increase to coverage.

Thanks!